src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -87,148 +87,152 @@ module LeafExpressionConverter =
                         if isg1 then minfo2.IsGenericMethod && gmd = minfo2.GetGenericMethodDefinition()
                         else minfo = minfo2
                      ) ->
-                Some (obj, (minfo2.GetGenericArguments() |> Array.toList), args)
+                Some (obj, minfo2, args)
             | _ -> None)
 
-
-    let (|SpecificCallToMethod|_|) (mhandle: System.RuntimeMethodHandle) =
+    let (|SpecificCallToMethod'|_|) (mhandle: System.RuntimeMethodHandle) =
```

Comment:
Please don't use `'` characters in names in FSharp.Core.  Also could you please minimize the diff in this file more?  There are a lot of changes and it's hard to see what's going on

Reply:
I think you can keep this as `SpecificCallToMethod` and have the on on [line 97](https://github.com/dotnet/fsharp/pull/11681/files#diff-69b099f4ca89531c437b4f5cbb1f31aa08d208aabdfda9f4efed06e7c9a342d3R97) be `SpecificCallToGenericMethod`?

Reply:
That would bring diff from `Query.fs`.

Reply:
That's ok. The overall change should still be clearer?

Reply:
I'm still generally not understanding what in particular requires such a large change in this file - 513 lines....  It's nearly all of the change.  And yet almost none of it seems to be "subtraction of two chars".  

I'm really happy if other bugs are being fixed, but all the fixes need to be documented in the PR and the diff minimal thanks